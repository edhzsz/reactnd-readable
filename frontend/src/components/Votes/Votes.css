.votes {
  float: left;
  margin: 10px;
}

.votes span {
  text-align: center;
}

.up-vote button {
  transform: rotate(180deg);
  box-shadow: 0 -3px 6px rgba(0,0,0,0.16), 0 -3px 6px rgba(0,0,0,0.23);
}