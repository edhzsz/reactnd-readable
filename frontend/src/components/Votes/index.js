import React from 'react';
import PropTypes from 'prop-types';
import './Votes.css'
import RoundButton from '../RoundButton';
import downArrowIcon from './icons/down_arrow.svg'

const Votes = (props) => {
  return (
    <div className="votes">
      <RoundButton type="secondary" className="up-vote" backgroundImage={downArrowIcon} size={props.size}>Up Vote</RoundButton>
      <span>{props.votes}</span>
      <RoundButton type="secondary" backgroundImage={downArrowIcon} size={props.size}>Down Vote</RoundButton>
    </div>
  );
}

Votes.propTypes = {
  size: PropTypes.number.isRequired,
  votes: PropTypes.number.isRequired,
  onUpvote: PropTypes.func,
  onDownvote: PropTypes.func
}

// Specifies the default values for props:
Votes.defaultProps = {
  size: 30
}

export default Votes;