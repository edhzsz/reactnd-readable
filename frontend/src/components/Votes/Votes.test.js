import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Votes from '.';

const wrapper = shallow(<Votes votes={2}/>);

describe('Component: Votes', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Votes votes={2}/>, div);
  });

  it('should match its snapshot with required props', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('should dispatch action when clicking upvote and downvote', () => {
    const onUpvoteButtonClick = jest.fn();
    const onDownvoteButtonClick = jest.fn();
    const wrapper = mount(<Votes votes={2} onUpvote={onUpvoteButtonClick} onDownvote={onDownvoteButtonClick}/>)

    wrapper.find('.upvote button').simulate('click')
    wrapper.find('.downvote button').simulate('click')

    expect(onUpvoteButtonClick.mock.calls.length).toBe(1);
    expect(onDownvoteButtonClick.mock.calls.length).toBe(1);
  });
});