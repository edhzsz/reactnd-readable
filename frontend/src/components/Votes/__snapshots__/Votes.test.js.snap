// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Votes should match its snapshot with required props 1`] = `
<div
  className="votes"
>
  <RoundButton
    backgroundImage="down_arrow.svg"
    className="upvote"
    size={30}
    type="secondary"
  >
    Up Vote
  </RoundButton>
  <span
    style={
      Object {
        "fontSize": 15,
      }
    }
  >
    2
  </span>
  <RoundButton
    backgroundImage="down_arrow.svg"
    className="downvote"
    size={30}
    type="secondary"
  >
    Down Vote
  </RoundButton>
</div>
`;
